.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+name: Unittests & Auto-publish
+
+# Allow to trigger the workflow manually (e.g. when deps changes)
+on: [push, workflow_dispatch]
+
+jobs:
+  unittest-job:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}-${{ matrix.python-version }}
+      cancel-in-progress: true
+
+    steps:
+    - uses: actions/checkout@v3
+
+    # Install deps
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - run: pip --version
+    - run: pip install -e .
+    - run: pip freeze
+
+    # Run tests
+    - name: Run core tests
+      run: python -m unittest discover --pattern "*_test.py" --start-directory "pathwaysutils/test"
```

Comment:
Does this call into absl test? Asking since it's using unittest.

Reply:
https://github.com/abseil/abseil-py/issues/138

> abseil doesn't have special logic for test discovery. Since absl.testing.absltest is built on top of unittest, if you just use absl.testing.absltest.TestCase over unittest.TestCase in your code, python -m unittest would also work.

Apparently fails if we are making use of some absl flags which we are not doing.